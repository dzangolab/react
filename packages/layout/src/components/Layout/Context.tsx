import { createContext, useContext, useState } from "react";

export type LayoutCotextType = {
  menuDesktopOpen: boolean;
  menuMobileOpen: boolean;
  setMenuDesktopOpen: (value: boolean) => void;
  setMenuMobileOpen: (value: boolean) => void;
};

export const LayoutContext = createContext<LayoutCotextType | null>(null);

type LayoutProviderProperties = {
  children?: React.ReactNode;
};

export const LayoutProvider = ({ children }: LayoutProviderProperties) => {
  const [menuDesktopOpen, setMenuDesktopOpen] = useState(true);
  const [menuMobileOpen, setMenuMobileOpen] = useState(false);

  return (
    <LayoutContext.Provider
      value={{
        menuDesktopOpen,
        menuMobileOpen,
        setMenuDesktopOpen,
        setMenuMobileOpen,
      }}
    >
      {children}
    </LayoutContext.Provider>
  );
};

export const useLayoutContext = () => {
  const context = useContext(LayoutContext);

  if (context === null) {
    throw new Error("LayoutProvider is required!");
  }

  return context;
};
