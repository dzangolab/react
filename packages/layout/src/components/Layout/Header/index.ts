export * from "./Header";
export * from "./HeaderMenu";
export * from "./HeaderTitle";
