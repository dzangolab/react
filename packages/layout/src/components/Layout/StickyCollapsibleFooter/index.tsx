import React from "react";

interface Properties {
  children?: React.ReactNode;
}

export const StickyCollapsibleFooter = ({ children }: Properties) => {
  return (
    <>
      <div className="footer-wrapper">
        <footer className="footer-content">{children}</footer>
      </div>
    </>
  );
};
