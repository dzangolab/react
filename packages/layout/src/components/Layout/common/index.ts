export * from "./Copyright";
export * from "./Logo";
export * from "./NavigationMenu";
export * from "./Toggle";
export * from "./Version";
