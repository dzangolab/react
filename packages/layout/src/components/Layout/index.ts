export * from "./Footer";
export * from "./Header";
export * from "./Layout";
export * from "./Sidebar";
export * from "./common";
export * from "./types";
