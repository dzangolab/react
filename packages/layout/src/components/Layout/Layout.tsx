import { useMediaQuery } from "@dzangolab/react-ui";

import { LayoutProvider, useLayoutContext } from "./Context";

type LayoutProperties = {
  children?: React.ReactNode;
  className?: string;
  fixed?: boolean;
};

const LayoutComponent = ({
  children,
  className,
  fixed = false,
}: LayoutProperties) => {
  const { menuDesktopOpen, menuMobileOpen } = useLayoutContext();
  const isSmallScreen = useMediaQuery("(max-width: 576px)");

  return (
    <div
      className={`dz-layout ${className || ""} ${
        fixed ? "fixed" : ""
      }`.trimEnd()}
      aria-expanded={isSmallScreen ? menuMobileOpen : menuDesktopOpen}
    >
      {children}
    </div>
  );
};

export const Layout = ({ children, className, fixed }: LayoutProperties) => {
  return (
    <LayoutProvider>
      <LayoutComponent className={className} fixed={fixed}>
        {children}
      </LayoutComponent>
    </LayoutProvider>
  );
};
