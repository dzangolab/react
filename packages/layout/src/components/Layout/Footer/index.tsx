import React from "react";

import { Copyright, Version } from "../common";

interface FooterProperties {
  children?: React.ReactNode;
}

export const Footer = ({ children }: FooterProperties) => {
  const renderContent = () => {
    return (
      <>
        <Copyright />
        <Version />
      </>
    );
  };
  return <footer>{children || renderContent()}</footer>;
};
