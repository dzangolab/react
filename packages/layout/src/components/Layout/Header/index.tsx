import { LocaleSwitcher } from "@dzangolab/react-i18n";
import React from "react";

import { Logo, NavigationMenu, ToggleMenuMobile } from "../common";
import { NavMenuType } from "../types";

interface HeaderProperties {
  children?: React.ReactNode;
  displayNavIcons?: boolean;
  navigationMenu?: NavMenuType;
  userMenu?: React.ReactNode; // TODO update to NavigationMenuItem[] when we have UserMenu component in layout package
}

export const Header = ({
  children,
  displayNavIcons,
  navigationMenu,
  userMenu,
}: HeaderProperties) => {
  const renderContent = () => {
    return (
      <>
        <Logo />
        <NavigationMenu
          navigationMenu={navigationMenu || []}
          displayIcons={displayNavIcons}
          horizontal
        />
        {userMenu}
        <LocaleSwitcher />
        <ToggleMenuMobile />
      </>
    );
  };
  return <header>{children || renderContent()}</header>;
};
