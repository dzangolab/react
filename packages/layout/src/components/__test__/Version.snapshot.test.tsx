import { AppConfig, configContext } from "@dzangolab/react-config";
import i18n from "@dzangolab/react-i18n";
import { render } from "@testing-library/react";
import { expect, test } from "vitest";

import { Version } from "../Layout";

const appConfig: AppConfig = {
  apiBaseUrl: "/",
  appPort: "20072",
  appTitle: "Dzango Skeletons",
  appVersion: "0.0.1",
  features: {
    showVersion: true,
  },
  i18n: {
    appendNamespaceToCIMode: true,
    debug: true,
    defaultNS: "app",
    fallbackLng: "en",
    supportedLngs: ["en", "fr"],
    react: {
      useSuspense: false,
    },
    resources: { en: {}, fr: {} },
  },
  websiteDomain: "//",
  copyright: {
    holder: "Dzango",
    url: "www.dzango.com",
  },
};

test("Component matches snapshot", () => {
  i18n(appConfig.i18n);

  const { container } = render(
    <configContext.Provider value={appConfig}>
      <Version version="1.0.0+local" />
    </configContext.Provider>,
  );

  expect(container).toMatchSnapshot();
});
