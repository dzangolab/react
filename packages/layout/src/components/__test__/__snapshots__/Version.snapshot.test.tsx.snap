// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<div>
  <div
    class="version"
  >
    1.0.0+local
  </div>
</div>
`;
